subprojects/core/src/main/java/org/gradle/api/services/internal/DefaultBuildServicesRegistry.java

```diff
@@ -132,26 +132,39 @@ public SharedResource forService(BuildServiceProvider<?, ?> service) {
 
     @Override
     public <T extends BuildService<P>, P extends BuildServiceParameters> Provider<T> registerIfAbsent(String name, Class<T> implementationType, Action<? super BuildServiceSpec<P>> configureAction) {
```

Comment:
refactored so the core logic would be shared by both conditional and unconditional registration methods